Misc/NEWS.d/next/Core and Builtins/2024-05-01-22-43-54.gh-issue-118473.QIvq9R.rst

```diff
@@ -0,0 +1 @@
+Fix set_asyncgen_hooks not to be partially set when raising TypeError
```

Comment:
Please make this proper ReST markup with links to the code for `set_asyncgen_hooks` and `TypeError`, and end the sentence with a period. (I can help with the markup but consulting other news files should give ample examples.)